pkg/kubectl/cmd/testing/fake.go

```diff
@@ -240,14 +245,58 @@ type TestFactory struct {
 	ClientConfigVal    *restclient.Config
 	CommandVal         string
 
+	tempConfigFile *os.File
+
 	UnstructuredClientForMappingFunc func(mapping *meta.RESTMapping) (resource.RESTClient, error)
 	OpenAPISchemaFunc                func() (openapi.Resources, error)
 }
 
 func NewTestFactory() *TestFactory {
-	return &TestFactory{
-		Factory: cmdutil.NewFactory(nil),
+	tf := &TestFactory{}
+
+	// specify an optionalClientConfig to explicitly use in testing
+	// to avoid polluting an existing user config.
+	tf.Factory = cmdutil.NewFactory(tf.defaultFakeClientConfig())
+	return tf
+}
+
+func (f *TestFactory) Cleanup() {
+	if f.tempConfigFile == nil {
+		return
+	}
+
+	os.Remove(f.tempConfigFile.Name())
+}
+
+func (f *TestFactory) defaultFakeClientConfig() clientcmd.ClientConfig {
+	loadingRules, err := f.newDefaultFakeClientConfigLoadingRules()
+	if err != nil {
+		panic(fmt.Sprintf("unable to create a fake client config: %v", err))
 	}
+
+	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+	fallbackReader := bytes.NewBuffer([]byte{})
+	overrides := &clientcmd.ConfigOverrides{ClusterDefaults: clientcmdapi.Cluster{Server: "http://localhost:8080"}}
+
+	clientConfig := clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, overrides, fallbackReader)
+	return clientConfig
+}
+
+func (f *TestFactory) newDefaultFakeClientConfigLoadingRules() (*clientcmd.ClientConfigLoadingRules, error) {
+	if f.tempConfigFile == nil {
```

Comment:
this doesn't seem to make sense.  Why don't you just return a loadingrules and the tempfile you created, then leave it to the `New` method to decide how to wire it.